compiler-rs/clients_schema_to_openapi/src/paths.rs

```diff
@@ -251,6 +251,43 @@ pub fn add_endpoint(
 
         let sum_desc = split_summary_desc(&endpoint.description);
 
+        // add the x-state extension for availability
+        let mut extensions = crate::availability_as_extensions(&endpoint.availability);
+
+        // add the x-codeSamples extension
+        if !request_examples.is_empty() {
+            let mut code_samples = vec![];
+            if let Some((_, first_example)) = request_examples.clone().first() {
+                if let Some(example) = first_example.as_item() {
+                    if let Some(description) = example.description.clone() {
```

Comment:
Could we use `summary` when `description` is not set?

Reply:
This feature does not need summaries or descriptions, all I'm doing here is parsing the description text with the hope that somewhere within it I can find the method and URL of the request. The only reason I'm using `description` is because that is how these examples are parsed.

The proper thing to do would be to expand these examples to include this information in a easier format, at which point there wouldn't be a need to use descriptions or summaries for anything.